docs/src/ext_dd.md

```diff
@@ -1 +1,85 @@
-# Extending DimensionalData
\ No newline at end of file
+# Extending DimensionalData
+
+Nearly everything in DimensionalData.jl is designed to be extensible.
+
+- `AbstractDimArray` are easily extended to custom array types. `Raster` or
+  `YAXArray` are examples from other packages.
+- `AbstractDimStack` are easily extended to custom mixed array dataset.
+    `RasterStack` or `ArViZ.Dataset` are examples.
+- `LookupArray` can have new types added, e.g. to `AbstractSampled` or
+  `AbstractCategorical`. `Rasters.Projected` is a lookup that knows
+  its coordinate reference system, but otherwise behaves as a regular
+  `Sampled` lookup.
+
+`dims` and `rebuild` are the key interface methods in most of these cases.
+
+## `dims`
+
+Objects extending DimensionalData.jl that have dimensions must return 
+a `Tuple` of constructed `Dimension`s from `dims(obj)`. 
+
+### `Dimension` axes
+
+Dimensions return from `dims` should hold a `LookupArray` or in some cases 
+just an `AbstractArray` (like wiht `DimIndices`). When attached to 
+mullti-dimensional objects, lookups must be the _same length_ as the axis 
+of the array it represents, and `eachindex(A, i)` and `eachindex(dim)` must 
```

Comment:
What are A and dim here? And there seems to be no eachindex(A, i) at least this also fails for normal Arrays.